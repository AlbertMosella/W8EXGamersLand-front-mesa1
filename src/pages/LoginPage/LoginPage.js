import LoginForm from "../../components/LoginForm/LoginForm";

const LoginPage = () => {
  return (
    <>
      <h1>User Login</h1>
      <LoginForm />
    </>
  );
};

export default LoginPage;
