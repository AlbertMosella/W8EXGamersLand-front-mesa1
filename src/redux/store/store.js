import { configureStore } from "@reduxjs/toolkit";
import gameReducer from "../features/gameSlice";
import userReducer from "../features/userSlice";
const store = configureStore({
  user: userReducer,
  games: gameReducer,
});

export default store;
