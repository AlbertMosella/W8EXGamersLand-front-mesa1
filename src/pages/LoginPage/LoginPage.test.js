import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import reactTestRenderer from "react-test-renderer";
import store from "../../redux/store/store";
import LoginPage from "./LoginPage";

describe("Given a LoginPage Component", () => {
  describe("When it's invoked", () => {
    test("Then it should render the text 'User Login'", () => {
      const expectedResult = "User Login";

      render(
        <Provider store={store}>
          <BrowserRouter>
            <LoginPage />
          </BrowserRouter>
        </Provider>
      );

      const receivedResult = screen.getByText(expectedResult);

      expect(receivedResult).toBeInTheDocument();
    });
  });

  describe("When it's rendered", () => {
    test("Then it alwais show the same structure", () => {
      const loginPage = reactTestRenderer.create(
        <Provider store={store}>
          <BrowserRouter>
            <LoginPage />
          </BrowserRouter>
        </Provider>
      );

      expect(loginPage).toMatchSnapshot();
    });
  });
});
