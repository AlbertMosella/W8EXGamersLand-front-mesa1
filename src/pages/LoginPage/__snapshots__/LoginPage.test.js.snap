// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a LoginPage Component When it's rendered Then it alwais show the same structure 1`] = `
Array [
  <h2>
    User Login
  </h2>,
  <div
    className="sc-bczRLJ eeDLXW"
  >
    <form
      noValidate={true}
      onSubmit={[Function]}
    >
      <label
        htmlFor="username"
      >
        Username
      </label>
      <input
        autoComplete="off"
        id="username"
        onChange={[Function]}
        type="text"
        value=""
      />
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        autoComplete="off"
        id="password"
        onChange={[Function]}
        type="password"
        value=""
      />
      <button
        className="form-button"
        type="submit"
      >
        Login
      </button>
    </form>
  </div>,
  <a
    className=""
    href="/register"
    onClick={[Function]}
  >
    Register
  </a>,
]
`;
